### PR TITLE
BIKE Round 2

### PR DESCRIPTION
**Pull request by drucker-nir**
This PR includes: 

1. Removing BIKE-2/3
2. Removing BIKE-1 Level-5
3. Renaming BIKE-1 Level-1/3 to BIKE-1-CPA
4. Adding BIKE-1 Level-1/3 with FO transformation and with negligible DFR.

### CODE REVIEW

File: src/kem/bike/additional/defs.h [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+ * The license is detailed in the file LICENSE.md, and applies to this file.
+ *
+ * Written by Nir Drucker and Shay Gueron
+ * AWS Cryptographic Algorithms Group.
+ * (ndrucker@amazon.com, gueron@amazon.com)
+ */
+
+#pragma once
+
+#include "../../../../config.h"
```
Comment (xvzcf) [inline src/kem/bike/additional/defs.h]: Is it possible to remove this dependency?

Comment (drucker-nir) [reply src/kem/bike/additional/defs.h]: Done

File: src/kem/bike/additional/measurements.h [hunk 2]
```diff
@@ -0,0 +1,103 @@
+/*
```
Comment (dstebila) [inline src/kem/bike/additional/measurements.h]: Do we need this file?

File: src/kem/bike/kem_bike.c [hunk 3]
```diff
@@ -2,290 +2,102 @@
...
-	kem->method_name = OQS_KEM_alg_bike3_l1;
-#if defined(OQS_KEM_BIKE_ADDITIONAL_IMPLEMENTATION)
+	kem->method_name = OQS_KEM_alg_bike1_l1_fo;
 	kem->alg_version = "Additional - 05/23/2018";
-#else
-	kem->alg_version = "Reference - 06/29/2018";
-#endif
 
 	kem->claimed_nist_level = 1;
 	kem->ind_cca = false;
```
Comment (dstebila) [inline src/kem/bike/kem_bike.c]: Should `kem->ind_cca = true` for the `_fo` variants?

File: src/kem/bike/kem_bike.c [hunk 4]
```diff
@@ -2,290 +2,102 @@
...
-	kem->method_name = OQS_KEM_alg_bike3_l3;
-#if defined(OQS_KEM_BIKE_ADDITIONAL_IMPLEMENTATION)
+	kem->method_name = OQS_KEM_alg_bike1_l3_fo;
 	kem->alg_version = "Additional - 05/23/2018";
-#else
-	kem->alg_version = "Reference - 06/29/2018";
-#endif
 
 	kem->claimed_nist_level = 3;
 	kem->ind_cca = false;
```
Comment (dstebila) [inline src/kem/bike/kem_bike.c]: Should `kem->ind_cca = true` for the `_fo` variants?

File: src/kem/kem.h [hunk 5]
```diff
@@ -31,24 +31,14 @@ extern "C" {
...
-/** Algorithm identifier for BIKE3-L5 KEM. */
-#define OQS_KEM_alg_bike3_l5 "BIKE3-L5"
+/** Algorithm identifier for BIKE1-L1-CPA KEM. */
+#define OQS_KEM_alg_bike1_l1_cpa "BIKE1-L1-CPA"
+/** Algorithm identifier for BIKE1-L3-CPA KEM. */
+#define OQS_KEM_alg_bike1_l3_cpa "BIKE1-L3-CPA"
+/** Algorithm identifier for BIKE1-L1-FO KEM. */
+#define OQS_KEM_alg_bike1_l1_fo "BIKE1-L1-FO"
+/** Algorithm identifier for BIKE1-L3-FO KEM. */
+#define OQS_KEM_alg_bike1_l3_fo "BIKE1-L3-FO"
```
Comment (dstebila) [inline src/kem/kem.h]: Also need to edit `OQS_KEM_algs_length` below.

**Comment by dstebila** [comment]
Note to self: before merging, update `OQS_KEM_algs_length` in templating and re-run copy-from-pqclean to ensure subsequent applications of copy-from-pqclean are clean.

**Comment by dstebila** [comment]
@drucker-nir To confirm my understanding, do the reference and additional implementations now have the same KATs?  If memory serves, with the Round 1 code there were different key sizes and KATs for the different implementations.

File: src/kem/bike/additional/measurements.h (hunk 2 above)
Comment (drucker-nir) [reply src/kem/bike/additional/measurements.h]: No, will remove.

**Comment by drucker-nir** [comment]
BIKE has wo implementations: the Reference implementation and the Additional implementation that have different (but equivalent) KATs. Previously, LibOQS used the Additional implementation for platforms that support AVX2 and the Reference implementation for other platforms. In Round-2 the Additional implementation supports all platforms. Therefore, we now need only one implementation and one set of KATs (i.e., this PR).

File: src/kem/bike/kem_bike.c (hunk 4 above)
Comment (Shay-Gueron) [reply src/kem/bike/kem_bike.c]: No. 
The _fo variants provide te FO flows and negligible DFR. 
Currently, we prefer not to claim IND-CCA security. 
In any case, the intention is to use this KEM only with ephemeral keys (for forward secrecy), and in such cases, an IND-CPA suffices. 
A paper with the relevant analysis is going to be posted soon on ePrint.

**Pull request merged**